packages/auth/amplify_auth_cognito/android/build.gradle

```diff
@@ -28,12 +28,12 @@ android {
     compileSdkVersion 33
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
```

Comment:
Seems like this could be an issue?

Reply:
From what I can tell, it is safe and maybe even [preferred](https://stackoverflow.com/a/69039059). I think this only requires that Java 11 or higher is installed on the user's machine and opts-in the library to use Java 11 features.

Flutter apps have the following by default in the app's `build.gradle`

```gradle
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    kotlinOptions {
        jvmTarget = '1.8'
    }
```

Even removing this, I was able to compile just fine. The only thing I can think is that it affects really old devices, but our integ tests should catch that as well.